Portscout/SiteHandler/GitHub.pm

```diff
@@ -101,38 +101,22 @@ sub GetFiles
 	# Extract project name from URL
 	if ($url =~ /https?:\/\/codeload\.github\.com\/(.+?)\/tar.gz\//) {
 		$projname = $1;
-	} elsif ($url =~ /https:\/\/github\.com\/(.+?)\/archive\//) {
+	}
+	elsif ($url =~ /https:\/\/github\.com\/(.+?)\/archive\//) {
 		$projname = $1;
-	} elsif ($url =~ /https:\/\/github.com\/downloads\/(.+)\//) {
+	}
+	elsif ($url =~ /https:\/\/github.com\/downloads\/(.+)\//) {
 		$projname = $1;
-	} else {
+	}
+	else {
 		_debug("Couldn't extract project name from URL $url");
 		return 0;
 	}
 
-	# GitHub Client ID & Secret to be appended to queries
-	# if they are set in settings
-	# https://developer.github.com/v3/#authentication
-	my $credentials = "";
-	if ($settings{github_client_id} && $settings{github_client_secret}) {
-		$credentials = "?client_id=$settings{github_client_id}&client_secret=$settings{github_client_secret}";
-	}
-
 	# See if there are any releases
-	my $query = 'https://api.github.com/repos/' . $projname . '/releases' . $credentials;
-	_debug("GET $query");
-	my $ua = LWP::UserAgent->new;
-	$ua->agent(USER_AGENT);
-	$ua->timeout($settings{http_timeout});
-
-	my $response = $ua->request(HTTP::Request->new(GET => $query));
-	if (!$response->is_success || $response->status_line !~ /^2/) {
-		_debug('GET failed: ' . $response->status_line);
-		return 0;
-	}
-
-	my $json = decode_json($response->decoded_content);
-	foreach my $release (@$json) {
+	my $releases = _call_github_api('/repos/' . $projname . '/releases')
```

Comment:
Removed adding credentials to the URL (as those are done through a token in the new _call_github_api call)